beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2440,6 +2440,15 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         proposer_slashing: ProposerSlashing,
     ) -> Result<ObservationOutcome<ProposerSlashing, T::EthSpec>, Error> {
         let wall_clock_state = self.wall_clock_state()?;
+
+        if let Some(event_handler) = self.event_handler.as_ref() {
```

Comment:
I'm not super familiar with these code paths, but shouldn't we only register the event once we've verified the slashing is `ObservationOutcome::New`? If it's invalid or `AlreadyKnown` then it would just be noise?

Reply:
This code block should be deleted, emitting the event in `import_proposer_slashing` covers both gossip and API `ObservationOutcome::New`

Reply:
updated in a3e42d31e1a75d050544502d658543ef6d7c6d97